target/linux/x86/base-files/lib/upgrade/platform.sh

```diff
@@ -80,7 +123,54 @@ platform_do_bootloader_upgrade() {
 }
 
 platform_do_upgrade() {
-	local diskdev partdev diff
+	local _alt altboot bootpart defboot diff diskdev partdev partdevalt part size start
+
+	#extract the boot sector from the image
+	get_image "$@" | dd of=/tmp/image.bs count=63 bs=512b 2>/dev/null
+
+	if is_efi_system; then
+		export_partdevice_label bootpart openwrt_boot \
+			&& export_partdevice_label partdev openwrt_rootfs \
+			&& export_partdevice_label partdevalt openwrt_rootfs_alt || {
+				echo "Unable to determine upgrade device"
```

Comment:
Use `>&2`.

Reply:
I'll add the redirects and make sure there is no difference with that `-$(CP)` line removed